app/javascript/src/Settings/components/AuthenticationSettingsFieldset.jsx

```diff
@@ -0,0 +1,47 @@
+// @flow
+
+import * as React from 'react'
+import { FormFieldset, FormLegend } from 'Form'
+import { FormCollection, TextInputGroup } from 'Settings/components/Common'
+import { OidcFieldset } from 'Settings/components/OidcFieldset'
+import type { FormSettingsProps, TypeItemProps } from 'Settings/types'
+
+const OIDC_AUTH_METHOD = 'oidc'
+const API_KEY_METHOD = '1'
+const APP_ID_KEY_METHOD = '2'
+
+type Props = {
+  isServiceMesh: boolean,
+  authenticationMethod: string,
+  apiKeySettings: FormSettingsProps,
+  appIdKeyPairSettings: FormSettingsProps[],
+  oidcSettings: {
+    basicSettings: TypeItemProps,
+    flowSettings: FormSettingsProps[],
+    jwtSettings: TypeItemProps
+  }
+}
+
+const AuthenticationSettingsFieldset = ({
+  isServiceMesh,
+  authenticationMethod,
+  apiKeySettings,
+  appIdKeyPairSettings,
+  oidcSettings
+}: Props) => {
+  const isOidc = authenticationMethod === OIDC_AUTH_METHOD
+  const isApiKey = authenticationMethod === API_KEY_METHOD
+  const isAppIdKey = authenticationMethod === APP_ID_KEY_METHOD
+  return (
+    <FormFieldset id='fieldset-AuthenticationSettings'>
+      { (!isServiceMesh || isOidc) && <FormLegend>Authentication Settings</FormLegend> }
+      { !isServiceMesh && isApiKey && <FormCollection collection={[apiKeySettings]} ItemComponent={TextInputGroup} legend='API KEY (USER_KEY) BASICS' /> }
+      { !isServiceMesh && isAppIdKey && <FormCollection collection={appIdKeyPairSettings} ItemComponent={TextInputGroup} legend='APP_ID AND APP_KEY PAIR BASICS' /> }
+      { isOidc && <OidcFieldset {...oidcSettings} isServiceMesh={isServiceMesh} /> }
```

Comment:
According with lines 37:40, is it possible to render an empty FormFieldSet? Namely when everything is false.